secret_store/src/key_server_cluster/admin_sessions/servers_set_change_session.rs

```diff
@@ -496,22 +503,20 @@ impl SessionImpl {
 				let master_node_id = message.master_node_id.clone().into();
 				if let Some(key_share) = self.core.key_storage.get(&key_id)? {
 					let version = message.version.clone().into();
-					if let Ok(key_version) = key_share.version(&version) {
-						let key_share_owners = key_version.id_numbers.keys().cloned().collect();
-						let new_nodes_set = data.new_nodes_set.as_ref()
-							.expect("new_nodes_set is filled during consensus establishing; change sessions are running after this; qed");
-						let local_plan = prepare_share_change_session_plan(
-							&self.core.all_nodes_set,
-							key_share.threshold,
-							&key_id,
-							version,
-							&master_node_id,
-							&key_share_owners,
-							new_nodes_set)?;
-
-						if local_plan.new_nodes_map.keys().collect::<BTreeSet<_>>() != master_plan.new_nodes_map.keys().collect::<BTreeSet<_>>() {
-							return Err(Error::InvalidMessage);
-						}
+					let key_share_owners = message.version_holders.iter().cloned().map(Into::into).collect();
+					let new_nodes_set = data.new_nodes_set.as_ref()
+						.expect("new_nodes_set is filled during consensus establishing; change sessions are running after this; qed");
```

Comment:
Can you explain why `expect()` here instead of returning an error?

Reply:
1) because I'm sure that the error won't ever happen. And I'm sure because...
2) ...the `ServerSetChange` session consists of two (omitting details) phases. The first one (`EstablishingConsensus`) is when master node prepares session structure (i.e. asking all other nodes - if they want to participate in the session, what shares do they have && what to do with these shares - either leave as is or add/remove). The second phase starts (`RunningShareChangeSessions`) after first phase is finished and it is the phase where shares are actually altered (new shares are added/old shares are removed). `on_initialize_share_change_session` is called during a second phase - it is checked at the beginning of the method. And `new_nodes_set` is initialized during first phase - on master node during session `initialize` call and on other nodes - when `UnknownSessionsRequest` message is received (that's the transition from phase to phase2).